src/Core/AndroidNative/maui/build.gradle

```diff
@@ -20,8 +21,8 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
```

Comment:
I would undo this line for now, we still build with Java source targeting 1.8, see for example:
* https://github.com/dotnet/android/commit/e517d535410a56a670a28b93ebf8a81c239ef92e